modules/c++/math.linear/tests/test_angle.cpp

```diff
@@ -0,0 +1,92 @@
+/* =========================================================================
+ * This file is part of math.linear-c++
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2014, MDA Information Systems LLC
+ *
+ * math.linear-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <TestCase.h>
+#include <import/math/linear.h>
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+// Explicitly recreating the Vector3D implementation of angle function
+double vec3DAngle(const math::linear::VectorN<3>& v1, const math::linear::VectorN<3>& v2)
```

Comment:
Suppose on one platform there was a bug in our actual implementation of the angle calculation (we divided by 0 or did something else weird and unsafe).  We wouldn't catch it with this approach since it would do the same wrong thing in this function and in the actual Vector implementation.  So rather than this, I'd pick 5-10 known cases and add these with explicit hard-coded expected results.  You could do that by just running your current implementation and printing out the current values and then using those as the hard-coded values to use.